DTAdmin/Subject/Subject/Controllers/SubjectTableViewController.swift

```diff
@@ -132,17 +132,18 @@ class SubjectTableViewController: UITableViewController, UISearchBarDelegate {
         return [delete, update]
     }
     
-    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        if let selectedSubject = self.selectedSubject {
-            selectedSubject(self.records[indexPath.row])
-            self.navigationController?.popViewController(animated: true)
-        } else {
-            if let wayToShowTests = UIStoryboard(name: "Subjects", bundle: nil).instantiateViewController(withIdentifier: "DetailSubject") as? DetailSubjectViewController
-            {
-                wayToShowTests.subject = self.records[indexPath.row]
-                self.navigationController?.pushViewController(wayToShowTests, animated: true)
-            }
-        }
+    func didTapShowTest(id: String) {
+        guard let wayToShowTestsForSubject = UIStoryboard(name: "Subjects", bundle: nil).instantiateViewController(withIdentifier: "TestForSubjectTableViewController") as? TestsForSubjectTableViewController else { return }
```

Comment:
IMO very strange name `wayToShowTestsForSubject`. Its really hard to follow. Better to rename it to ` subjectTestsTableViewController`.